File: Keep-Multiplying-Found-Values-by-Two.js
/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function(nums, o) {
    const set = new Set(nums);

    while(set.has(o)){
        o = o * 2;
    }
    return o;
};

File: Keep-Multiplying-Found-Values-by-Two.cpp
class Solution {
public:
    int findFinalValue(vector<int>& nums, int o) {
        unordered_set<int> n(nums.begin(),nums.end());
        
        while(n.count(o)){
            o = o * 2;
        }
        return o;
    }
};